specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2018-02-01-preview/containerregistry_build.json

```diff
@@ -0,0 +1,2136 @@
+        }
+      }
+    },
+    "PlatformProperties": {
+      "description": "The platform properties against which the build has to happen.",
+      "required": [
+        "osType"
+      ],
+      "type": "object",
+      "properties": {
+        "osType": {
+          "description": "The operating system type required for the build.",
+          "enum": [
+            "Windows",
+            "Linux"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "OsTypes",
+            "modelAsString": true
+          }
+        },
+        "cpu": {
+          "format": "int32",
+          "description": "The CPU configuration in terms of number of cores required for the build.",
+          "type": "integer"
+        }
+      }
+    },
+    "BuildTrigger": {
+      "description": "Base properties for any build trigger.",
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "The type of the trigger.",
+          "enum": [
+            "ImageTrigger"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "TriggerType",
+            "modelAsString": true
+          }
+        }
+      },
+      "discriminator": "type"
+    },
+    "BuildUpdateParameters": {
+      "description": "The set of build properties that can be updated.",
+      "type": "object",
+      "properties": {
+        "isArchiveEnabled": {
+          "description": "The value that indicates whether archiving is enabled or not.",
+          "type": "boolean"
+        }
+      }
+    },
+    "BuildLogParameters": {
+      "description": "The parameters for requesting build logs.",
+      "required": [
+        "logType"
+      ],
+      "type": "object",
+      "properties": {
+        "logType": {
+          "description": "The type of build log.",
+          "enum": [
+            "RawText"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "BuildLogType",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "BuildLogResult": {
+      "description": "The result of get build logs.",
+      "required": [
+        "logType",
+        "logLink"
+      ],
+      "type": "object",
+      "properties": {
+        "logType": {
+          "description": "The type of build log.",
+          "enum": [
+            "RawText"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "BuildLogType",
+            "modelAsString": true
+          }
+        },
+        "logLink": {
+          "description": "The link to the build log.",
+          "type": "string"
+        }
+      }
+    },
+    "QueueBuildRequest": {
+      "description": "The queue build request parameters.",
+      "required": [
+        "imageName",
+        "sourceLocation",
+        "buildParameters"
+      ],
+      "type": "object",
+      "properties": {
+        "imageName": {
```

Comment:
Compared with `ImageDescriptor.imageName`, this `imageName` seems to make more sense.